.github/workflows/bindings_haskell.yml

```diff
@@ -70,3 +70,54 @@ jobs:
           path: |
             ~/.cabal/store
             bindings/haskell/dist-newstyle
+
+  package:
+    runs-on: ubuntu-latest
+    if: "startsWith(github.ref, 'refs/tags/')"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Haskell toolchain (ghc-9.2.8)
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
+          ghcup install ghc 9.2.8 --set
+          ghcup install cabal --set
+          cabal update
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup
+      - name: Restore haskell cache
+        uses: actions/cache/restore@v3
+        with:
+          key: ${{ runner.os }}-haskell-${{ hashFiles('**/*.cabal', '**/Setup.hs') }}
+          path: |
+            ~/.cabal/store
+            bindings/haskell/dist-newstyle
+          restore-keys: ${{ runner.os }}-haskell-
+      - name: Package
+        working-directory: "bindings/haskell"
+        run: |
+          cargo package --target-dir target
+          cd target/package
+          tar xf opendal-hs-*.crate --strip-components=1
+          cabal sdist
+      - name: Publish candidate to Hackage
+        if: "contains(github.ref, 'rc')"
+        run: |
+          cabal upload -t ${{ secrets.HACKAGE_TOKEN }} bindings/haskell/target/package/dist-newstyle/sdist/*.tar.gz
```

Comment:
We can't release anything until we have passed vote (even for rc tags). Please only upload release for our official tags.

Reply:
removed in new commit